lib/mwoffliner.lib.js

```diff
@@ -911,6 +927,99 @@ module.exports = {
                 }
             }
 
+            function storeDependencies( parsoidDoc, articleId, finished ) {
```

Comment:
> This is the only point where I see a real problem. To know if a module is there or not, this makes a disk access, which is slow.
> 
> What I have proposed is that instead of this, the track of modules could be handled through a a simple global hash, keeping the track of what module has already been downloaded (or need to be downloaded).

This is not true anymore: the code now write/reads in Redis. 
See https://github.com/Skylsmoi/mwoffliner/blob/1475fda27411fb7d4556c733c0ae01453da046dc/lib/mwoffliner.lib.js

> If you do that, then it might be cleaner - not faster - to just then download the modules sources at the end.

I agree with you about spliting "detection of dependencies" and "downloading dependencies".

> If that point is fixed, then we could probably go ahead with the merge as the other one are minors - or cleared.

@Skylsmoi , let us know when the code is splited.